### PR TITLE
feat(vouching): vouching timeout for challenged vouchees

### PR DESCRIPTION
**Pull request by bilinkis**
As discussed on the PoH governance forum, vouchers that had a vouchee challenged and removed in the last 60 days shouldn't be able to vouch. This will help mitigate vouch and challenge attacks. 
For any enquiry on this modifications I can be reached on Telegram (https://t.me/nicobilinkis)

### CODE REVIEW

File: _pages/profile/[id]/submission-details-card/index.js [hunk 1]
```diff
@@ -133,8 +136,36 @@ export default function SubmissionDetailsCard({
...
+    const latestChallengedVouchee = disputedVouches?.sort((a, b) =>
+      a.requests[0].challenges[0].creationTime <
+      b.requests[0].challenges[0].creationTime
+        ? 1
+        : -1
+    );
+
+    const challengeTimestamp =
+      latestChallengedVouchee[0].requests[0].challenges[0].creationTime * 1000;
+    const timestamp = challengeTimestamp + 5200000000 > Date.now();
```
Comment (0x676f) [inline _pages/profile/[id]/submission-details-card/index.js:164]: How did you calculate the number `5200000000`?

Comment (bilinkis) [reply _pages/profile/[id]/submission-details-card/index.js:164]: It’s the approximate amount of seconds that there are in 60 days.

Comment (0x676f) [reply _pages/profile/[id]/submission-details-card/index.js:164]: @bilinkis Then I suggest you replace the number with `60*24*60*60` (for increased readability):

```suggestion
    const timestamp = challengeTimestamp + 60*24*60*60 > Date.now();
```

Comment (bilinkis) [reply _pages/profile/[id]/submission-details-card/index.js:164]: It should be also multiplied by 1000 given that I the output of Date.now is in milliseconds. But it's a good catch. I'll commit that now.

**Comment by fnanni-0** [comment]
I don't think this is a good idea. I imagine that users who vouch-challenge know the system well and might have the technical skills to either interact with the smart contract directly or run the previous version of the UI locally. Vouching for someone can't be really prevented from the UI.

Best case scenario, I think this change would stop some vouch-challengers while the most experienced ones will continue to do it with less competitors around.

**Comment by eccentricexit** [comment]
I disagree, I believe the vouch challengers are not so skilled or ate least good portion of them would not go through the hassle of working around the issue to vouch challenge.

Even if it there is just one unskilled vouch-challenger it is still worth it to raise the costs of the attack

**Comment by fnanni-0** [comment]
I don't have a strong opinion about this. 

For the record, for the vouch-challenge issue I think time would be better spent on solutions that tackle the root of the problem, such as:
1. Improving the onboarding. For example, I believe that adding a preview page with a bunch of checkboxes will help users a lot.
1. Add a fine to users who vouched for challenged profiles.

**Comment by luiandresgonzalez** [comment]
As I mentioned in Telegram:
There is zero risk in adding this and some marginal benefit. It could prevent lousy (not necessarily bad) actors.
I still do not understand who and how decides if these PRs are implemented or not.

**Comment by JuanuMusic** [comment]
> 2. Add a fine to users who vouched for challenged profiles.
@fnanni-0 
How could this be implemented without a change on the contract?

Implementing it on the UI would be just a FE solution and still allow tech-saavy users to vouchallenge

**Comment by alcercu** [comment]
I do think it's a bad idea since there are non-negligible chances that at least one vouch-challenger knows how to do it without the frontend (or running locally an older version of it). By removing their competition we are increasing the overall number of deposits that the tech-savvy vouch-challenger will get, thus increasing the incentive to be looking for these opportunities. It would be more effective to improve the onboarding process.
An alternative would be to publicly flag these profiles, with a warning sign and red text, so people can know who is vouch-challenging and there is some kind of public shame in doing so because you will be marking your identity.

**Comment by fnanni-0** [comment]
> 2. Add a fine to users who vouched for challenged profiles.

I was thinking of a change to the policy. I'm not sure if I'm a fan of it, but it's possible. Here's an example of how it could work.

- If you vouched for an Invalid Submission, pay the SUBMITTER an amount equal to the juror fees to be able to reapply your submission in the future. This way the voucher shares part of losses with the submitter, the vouch-challenge behavior will unlikely be profitable and honest vouchers are incentivized to be more careful.
- If you vouched for a Deceased, Fake or Duplicate, pay the CHALLENGER an amount equal to the juror fees to be able to reapply your submission in the future. This way challengers have an extra incentive to find this hard type of frauds and the voucher gets extra penalized.

A simple contract similar to this one https://github.com/Proof-Of-Humanity/Proof-Of-Humanity/blob/master/contracts/Finer.sol could be used to track fine payments. Rejecting reapplications of users who haven't paid their fines would be enforced by specifying this in the policy.

Note that these vouchers can choose to ignore or delay the fine, but at the expense of being ostracized from POH after the submission has expired. This also increases the risk of selling/delegating your identity, as it could be used for vouch-challenging or for malicious vouching.

**Comment by luiandresgonzalez** [comment]
There is an unanswered question in this thread: who and how decides what PR
is to be approved?

**Comment by bilinkis** [comment]
> An alternative would be to publicly flag these profiles, with a warning sign and red text, so people can know who is vouch-challenging and there is some kind of public shame in doing so because you will be marking your identity.
This was already included on this PR, but I decided to separate that feature onto another branch given that implementing that shouldn't require much discussion. The discussion on the timeout feature can continue here, and the red avatars can be found on a new Pull request #488.

**Comment by fnanni-0** [comment]
> There is an unanswered question in this thread: who and how decides what PR is to be approved?

Anyone in the Proof-Of-Humanity repo team can merge a PR. Usually we wait for at least someone in the team to review before merging. If there are any objections regarding a feature, we discuss them with the rest team.

**Comment by luiandresgonzalez** [comment]
1. Is the list of members of the repo team known? 2. How do they decide?


*Luis Andrés Gonzalez Gracia  *
 *luis.andres.gonzalez |  @gonzaluisandres
<http://www.twitter.com/gonzaluisandres> |  luisandresgonzalez
<http://www.linkedin.com/in/luisandresgonzalez>* |
* <http://www.linkedin.com/in/luisandresgonzalez>+1 639-994-9701
<https://api.whatsapp.com/send?phone=16399949701>*

**Comment by 0x676f** [comment]
> 1. Is the list of members of the repo team known?

@luiandresgonzalez Not publicly. This information is only available for those who have been added as members of the GitHub organization.

However, we could make the member list public by adding them to a `CODEOWNERS` file: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

**Comment by clesaege** [comment]
This issue is not possible at the frontend level.
It doesn't make sense to make this frontend worse for users on purpose (as someone can just fork it and remove this limitation), so I'll close the issue.

**Pull request closed without merging**